src/common/filesystembase.cpp

```diff
@@ -550,6 +560,21 @@ bool FileSystem::isChildPathOf(const QString &child, const QString &parent)
         // clear trailing slashes etc
         || QString::compare(QDir::cleanPath(parent), QDir::cleanPath(child), sensitivity) == 0);
 }
+
+QString OCSYNC_EXPORT FileSystem::createPortableFileName(const QFileInfo &path, int reservedSize)
+{
+    // remove leading and trailing whitespace
+    QString tmp = path.fileName().trimmed();
+    // limit size to fileNameMaxC
+    tmp.resize(std::min(tmp.size(), fileNameMaxC - reservedSize));
+    // remove eventual trailing whitespace after the resize
+    tmp = tmp.trimmed();
+    // replace non portable characters
+    for (auto c : IllegalFilenameCharsWindows) {
```

Comment:
What about other OSes?

Also, I think it makes more sense to use an allowlist than an excludelist approach. Is that a viable option?

Reply:
The idea of the function is to create a portable file name, which means its the same on all platforms.

Reply:
If it applies to all platforms, you may want to change the name of that variable.

Reply:
The var is a list of characters that are illegal on Windows, it is used in different locations.
In this function it is used to create a portable file name.